docs/source/Objective_Functions/Objective_Function.md

```diff
@@ -1,25 +1,203 @@
+## Objective Function and Local Optimization Function Format
+TetraOpt requires the following way of communication for its objective and local optimization function.
+1. A communication channel (using the tq42 API)
+2. An https endpoint
 
+
+An example of TetraOpt parameters using the `communication channel`. Notice the
+`objective_function_channel_id` and `local_optimizer_channel_id` parameter:
+
+```
+objective_func_channel = await Channel.create(client=client)
+local_opt_channel = await Channel.create(client=client)
+
+tetra_opt_parameters = {
+    "dimensionality": 2,
+    "iteration_number": 2,
+    "maximal_rank": 4,
+    "points_number": 1,
+    "quantization": False,
+    "tolerance": 0.0010000000474974513,
+    "lower_limits": [0, 0],
+    "upper_limits": [9, 9],
+    "grid": [10, 10],
+     "objective_function_channel_id": objective_func_channel.id
+     #local_optimizer_channel_id parameter is optional
+    "local_optimizer_channel_id": local_opt_channel.id
+}
+```
+
+An example of TetraOpt parameters using `an https endpoint`. Notice the
+`objective_function` and `local_optimizer` parameter:
+```
+tetra_opt_parameters = {
+    "dimensionality": 2,
+    "iteration_number": 2,
+    "maximal_rank": 4,
+    "points_number": 1,
+    "quantization": False,
+    "tolerance": 0.0010000000474974513,
+    "lower_limits": [0, 0],
+    "upper_limits": [9, 9],
+    "grid": [10, 10],
+    'objective_function':'http://34.32.169.11:8000/test_func_eval/Ackley/',
+     #local_optimizer parameter is optional  
+    "local_optimizer": "http://34.32.169.11:8000/local_optimization/Ackley/",
+    "polling": {"initial_delay": 1.0, "retries": 100, "delay": 1.0, "backoff_factor": 1.1}
+}
+```
+
+### 1. To use the communications channel:
+The channel uses objects of the following classes to send and receive information:
+1. `Ask`
+2. `Tell`
+
+It also need the following class to stream the information from/to TetraOpt
+1. `Channel`
+
+We can import them from the tq42 library in our Python script using the following:
+
+`from tq42.channel import Channel, Ask, Tell`
+
+  Python types for Ask and Tell parameters:
+  - `ask.headers` -> list of strings
+  - `ask.parameter.values` -> list of floats
+  - `tell.results` -> list of floats
+  - `tell.candidates.values` -> list of floats
+  
+
+
+#### The `Ask` Class
+An example of an `Ask` object values passed by TetraOpt:
+```
+{
+    "parameters": [
+        {"values": [1, 3]},
+        {"values": [2, 7]}
+    ],
+    "headers": ["h0", "h1"]
+}
+```
+
+#### The `Tell` Class 
+An example of `Tell` object values for an objective and local optimization function:
+```
+{
+    "parameters": [
+        {"values": [1, 3]},
+        {"values": [2, 7]}
+    ],
+    "headers": ["h0", "h1"],
+    "results": [6.59359884, 7.86938667],
+    #parameter candidates is only used for local optimization function
+    "candidates": [
+        {"values": [-8.18565e-09, -8.18565e-09]}
```

Comment:
is this really only 1 candidate or are you getting 2 in this example? Not sure just now.